src/Control/Lens.hs

```diff
@@ -98,3 +125,803 @@ import Control.Lens.Zoom
 #ifdef HLINT
 {-# ANN module "HLint: ignore Use import/export shortcut" #-}
 #endif
+
+{- $tutorial
+    This @lens@ tutorial targets Haskell beginners and assumes only basic
+    familiarity with Haskell.  By the end of this tutorial you should:
+
+    * understand what problems the @lens@ library solves,
+
+    * know when it is appropriate to use the @lens@ library,
+
+    * be proficient in the most common @lens@ idioms,
+
+    * understand the drawbacks of using lenses, and:
+
+    * know where to look if you wish to learn more advanced tricks.
+-}
+
+-- $motivation
+--
+--     The simplest problem that the @lens@ library solves is updating deeply
+--     nested records.  Suppose you had the following nested Haskell data types:
+-- 
+-- > data Atom = Atom { _element :: String, _point :: Point }
+-- >
+-- > data Point = Point { _x :: Double, _y :: Double }
+-- 
+--     If you wanted to increase the @x@ coordinate of an `Atom` by one unit, you
+--     would have to write something like this in Haskell:
+-- 
+-- > shiftAtomX :: Atom -> Atom
+-- > shiftAtomX (Atom e (Point x y)) = Atom e (Point (x + 1) y)
```

Comment:
You can also write this as `shiftAtomX atom = atom { _point = _point atom { _x = _x (_point atom) + 1 } }`, so at least the drawback "... the more fields you add ..." is not quite right, because this expression doesn't depend on the number of fields.